src/main/java/com/github/thundermarket/thundermarket/service/ProductService.java

```diff
@@ -0,0 +1,34 @@
+package com.github.thundermarket.thundermarket.service;
+
+import com.github.thundermarket.thundermarket.domain.Product;
+import com.github.thundermarket.thundermarket.repository.ProductRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class ProductService {
+
+    private final ProductRepository productRepository;
+
+    public ProductService(ProductRepository userRepository) {
+        this.productRepository = userRepository;
+    }
+
+
+    public List<Product> getProductList() {
+        return productRepository.findAll();
+    }
+
+    public Product addProduct(Product product) {
+        return productRepository.save(product);
+    }
+
+    public Product editProduct(Product product) {
+        return productRepository.update(product);
+    }
+
+    public void deleteProduct(Long id) {
+        productRepository.delete(id);
+    }
```

Comment:
optional한 코멘트는 p5; 라고 앞에 붙일게요.
p5; 이미 파라미터가 product임을 알려주고 있어서 메소드 명에 따로 안들어가도 될 것 같아요.
```suggestion
    public Product add(Product product) {
        return productRepository.save(product);
    }

    public Product edit(Product product) {
        return productRepository.update(product);
    }

    public void delete(Long id) {
        productRepository.delete(id);
    }
```

Reply:
특정 모델을 수정할때 edit 보다는 modify, update 표현을 더 많이 써요.

Reply:
8c4db5e8eca563a33bba90152c913fe4937ae2d3

Repository에서 update()를 쓰고 있어서 서비스에서는 edit() 이라고 곂치지 않게 지어봤는데..
오히려 직관적인 이름을 방해한 면이 있었네요.

Repository interface에서 save가 add, update 두가지 역할을 포함하게 되면서
여기에서는 update를 사용하면 될 것 같습니다.

제안 주신 깔끔한 이름도 읽기 쉬운 것 같아서 반영하였습니다.